java/ql/src/experimental/Security/CWE/CWE-020/Log4jJndiInjection.ql

```diff
@@ -19,140 +19,146 @@ import semmle.code.java.dataflow.FlowSources
-        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object,Object,Object);;Argument[0..5];logging",
-        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object,Object,Object,Object);;Argument[0..6];logging",
-        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object,Object,Object,Object,Object);;Argument[0..7];logging",
-        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object,Object,Object,Object,Object,Object);;Argument[0..8];logging",
-        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object,Object,Object,Object,Object,Object,Object);;Argument[0..9];logging",
-        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object,Object,Object,Object,Object,Object,Object,Object);;Argument[0..10];logging",
-        "org.apache.logging.log4j;LogBuilder;true;log;(String,Supplier[]);;Argument[0..1];logging",
-        "org.apache.logging.log4j;LogBuilder;true;log;(Supplier);;Argument[0];logging"
+        "org.apache.logging.log4j;LogBuilder;true;log;(CharSequence);;Argument[0];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(Message);;Argument[0];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(Object);;Argument[0];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(String);;Argument[0];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object[]);;Argument[0..1];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object);;Argument[0..1];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object);;Argument[0..2];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object);;Argument[0..3];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object,Object);;Argument[0..4];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object,Object,Object);;Argument[0..5];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object,Object,Object,Object);;Argument[0..6];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object,Object,Object,Object,Object);;Argument[0..7];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object,Object,Object,Object,Object,Object);;Argument[0..8];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object,Object,Object,Object,Object,Object,Object);;Argument[0..9];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(String,Object,Object,Object,Object,Object,Object,Object,Object,Object,Object);;Argument[0..10];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(String,Supplier[]);;Argument[0..1];log4j",
+        "org.apache.logging.log4j;LogBuilder;true;log;(Supplier);;Argument[0];log4j",
+        // org.apache.logging.log4j.ThreadContet
```

Comment:
There are also [`CloseableThreadContext`](https://logging.apache.org/log4j/2.x/log4j-api/apidocs/org/apache/logging/log4j/CloseableThreadContext.html) and [`CloseableThreadContext.Instance`](https://logging.apache.org/log4j/2.x/log4j-api/apidocs/org/apache/logging/log4j/CloseableThreadContext.Instance.html), in case you want to cover them as well.

Reply:
Good catch @Marcono1234, thanks. See #7435.